pages/advanced/api-safe-transaction-service/guides/api.md

```diff
@@ -0,0 +1,627 @@
+
+```bash
+curl -X 'POST' \
+'https://safe-transaction-sepolia.safe.global/api/v1/messages/0xcf2e6b1e26e6930e14bebf120ffffffffffffffffffffb484a787137201ab0df/signatures/' \
+-H 'accept: application/json' \
+-H 'Content-Type: application/json' \
+-d '{
+    "signature": "0xc8bff3e71314f8b79f6e31ae38c5efba1194e61e8ba6f35742f73af4ffffffffffffffffffffe5f84088ba29f51c44ddee87af26ca7d71e8f008e3bcfc25f1d31c"
+}'
+```
+
+#### Using safe-core-sdk
+
+```typescript
+import { ethers } from 'ethers'
+import SafeApiKit, { AddMessageProps } from '@safe-global/api-kit'
+import Safe, { EthersAdapter, hashSafeMessage } from '@safe-global/protocol-kit'
+
+async function createMessageAndSignByOwners() {
+  // Instantiate an EthAdapter for Owner A
+  const ethProvider = new ethers.JsonRpcProvider(config.RPC_URL)
+
+  const ownerA = new ethers.Wallet(config.OWNER_A_PRIVATE_KEY, ethProvider)
+  const ethAdapterOwnerA = new EthersAdapter({
+    ethers,
+    signerOrProvider: ownerA
+  })
+
+  // Initialize the Protocol Kit for Owner A
+  const protocolKitOwnerA = await Safe.create({
+    ethAdapter: ethAdapterOwnerA,
+    safeAddress: config.SAFE_ADDRESS
+  })
+
+  const rawMessage: string = 'A Safe Message - ' + Date.now()
+
+  // Create Safe Message
+  const safeMessage = protocolKitOwnerA.createMessage(rawMessage)
+
+  console.log('Message created: %O', safeMessage)
+
+  const signedMessageOwnerA = await protocolKitOwnerA.signMessage(safeMessage)
+
+  console.log(
+    'Message signed by Owner A: %s',
+    signedMessageOwnerA.encodedSignatures()
+  )
+
+  // Initialize the Api Kit
+  const apiKit = new SafeApiKit({
+    chainId: 11155111n
+  })
+
+  // Add Message to Tx service
+  const messageProps: AddMessageProps = {
+    message: rawMessage,
+    signature: signedMessageOwnerA.encodedSignatures()
+  }
+
+  apiKit.addMessage(config.SAFE_ADDRESS, messageProps)
+
+  console.log('Message proposed to Safe Tx service')
+
+  // Instantiate an EthAdapter for Owner B
+  const ownerB = new ethers.Wallet(config.OWNER_B_PRIVATE_KEY, ethProvider)
+  const ethAdapterOwnerB = new EthersAdapter({
+    ethers,
+    signerOrProvider: ownerB
+  })
+
+  // Initialize the Protocol Kit for Owner B
+  const protocolKitOwnerB = await Safe.create({
+    ethAdapter: ethAdapterOwnerB,
+    safeAddress: config.SAFE_ADDRESS
+  })
+
+  // Sign Blockchain Message by Owner B
+  const signedMessageOwnerB = await protocolKitOwnerB.signMessage(safeMessage)
+
+  console.log(
+    'Message signed by Owner B: %s',
+    signedMessageOwnerB.encodedSignatures()
+  )
+
+  // Add signature of owner B in Safe Tx service
+  const safeMessageHash = await protocolKitOwnerB.getSafeMessageHash(
+    hashSafeMessage(rawMessage)
+  )
+  const ownerBAddress = (await ethAdapterOwnerB.getSignerAddress()) || '0x'
+  await apiKit.addMessageSignature(
+    safeMessageHash,
+    signedMessageOwnerB.getSignature(ownerBAddress)?.data || '0x'
+  )
+
+  console.log('Message Owner B signature added to Tx service.')
+}
+```
```

Comment:
No Python example?

Reply:
I just added it, I hadn't done because it existed [here](https://docs.safe.global/advanced/api-service-architecture/safe-transaction-service#offchain-messages), but yes, you are right and it is better to add it. Maybe we can remove the above example if we refactor the Tx service page.

Reply:
Nice, thanks!